nailgun/entities.py

```diff
@@ -6912,3 +6912,112 @@ def update(self, fields=None):
+            'organization': entity_fields.OneToOneField(Organization)
+        }
+        self._meta = {
+            'api_path': 'foreman_virt_who_configure/api/v2/configs',
+            'server_modes': ('sat', 'sam'),
+        }
+        super(VirtWhoConfig, self).__init__(server_config, **kwargs)
+
+    def path(self, which=None):
+        """Extend ``nailgun.entity_mixins.Entity.path``.
+
+        The format of the returned path depends on the value of ``which``:
+
+        deploy_script
+            /foreman_virt_who_configure/api/v2/configs/:id/deploy_script
+
+        ``super`` is called otherwise.
+
+        """
+        if which in ('deploy_script'):
+            return '{0}/{1}'.format(
+                super(VirtWhoConfig, self).path(which='self'), which)
+        return super(VirtWhoConfig, self).path(which)
+
+    def create_payload(self):
+        """
+        Wraps config in extra dict
+        :return:
+        """
+        return {'foreman_virt_who_configure_config':
+                super(VirtWhoConfig, self).create_payload()}
+
+    def deploy_script(self, synchronous=True, **kwargs):
+        """Helper for Config's deploy_script method.
+
+        :param synchronous: What should happen if the server returns an HTTP
+            202 (accepted) status code? Wait for the task to complete if
+            ``True``. Immediately return the server's response otherwise.
+        :param kwargs: Arguments to pass to requests.
+        :returns: The server's response, with all JSON decoded.
+        :raises: ``requests.exceptions.HTTPError`` If the server responds with
+            an HTTP 4XX or 5XX message.
+
+        """
+        kwargs = kwargs.copy()  # shadow the passed-in kwargs
+        kwargs.update(self._server_config.get_client_kwargs())
+        response = client.get(self.path('deploy_script'), **kwargs)
+        return _handle_response(response, self._server_config, synchronous)
+
+    def show(self, synchronous=True, **kwargs):
+        """Helper for Config's show method.
+
+        :param synchronous: What should happen if the server returns an HTTP
+            202 (accepted) status code? Wait for the task to complete if
+            ``True``. Immediately return the server's response otherwise.
+        :param kwargs: Arguments to pass to requests.
+        :returns: The server's response, with all JSON decoded.
+        :raises: ``requests.exceptions.HTTPError`` If the server responds with
+            an HTTP 4XX or 5XX message.
+
+        """
+        kwargs = kwargs.copy()  # shadow the passed-in kwargs
+        kwargs.update(self._server_config.get_client_kwargs())
+        response = client.get(self.path('self'), **kwargs)
+        return _handle_response(response, self._server_config, synchronous)
+
+    def read(self, entity=None, attrs=None, ignore=None, params=None):
+        """
+        Override :meth:`nailgun.entity_mixins.EntityReadMixin.read` to ignore
+        the ``hypervisor_password``
```

Comment:
Why ignore `hypervisor_password`?

Reply:
Because its a password, it can't be queried (you get a 405 or something similar). Overriding this method ensures we don't try to query that field.